cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -45,6 +54,31 @@ using internal::checked_cast;
 
 namespace engine {
 
+Status WriteParquetData(const std::string& path,
+                        const std::shared_ptr<fs::FileSystem> file_system,
+                        const std::shared_ptr<Table> input) {
+  EXPECT_OK_AND_ASSIGN(auto buffer_writer, file_system->OpenOutputStream(path));
```

Comment:
Generally a test method should either return a `Status` and not have any calls to `EXPECT_...` or it should return a value / void.

A good rule of thumb is:

Test case - Use ASSERT_, return void
Test helper method, used only within one file - Use ASSERT_, return void or use EXPECT_, return value
Test helper method, shared by multiple test files - Use ARROW_, return Status or Result

Reply:
Thank you for pointing this out. I will update accordingly.